### DISCUSSION TITLE
How to customize max_part_size max_file_size in FastAPI

### DISCUSSION THREAD

**Discussion by panla** [fastapi/fastapi#12978] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

server

```python
@app.post('/files')
async def upload(
    file: Optional[UploadFile] = File(None),
    encodefile: Optional[str] = Form(None)

):
    pass

```

req

```python

with open('a.mp4', 'rb) as f:
    b_data = f.read()
    b64_data = base64.b64encode(b_data).decode('utf-8')

file_data = {
    'file': b_data
}
form_data = {
    'encodeFile': b64_data
}

response = requests.post('http://127.0.0.1:8000/api/upload/files', data=multipart_data, files=file_data)

response:
    400, {"detail":"Part exceeded maximum size of 1024KB."}
```

### Description

in starlette `starlette\formparsers.py`

```python
class MultiPartParser:
    max_file_size = 1024 * 1024  # 1MB
    max_part_size = 1024 * 1024  # 1MB
```

How to customize max_part_size max_file_size in FastAPI

### Operating System

Linux, Windows

### Operating System Details

windows: 11
debian: 12

### FastAPI Version

fastapi==0.115.5

### Pydantic Version

pydantic==2.10.1

### Python Version

3.11

**Comment by luzzodev** [comment]
Hi @panla this is duplicate of https://github.com/fastapi/fastapi/discussions/12961#discussioncomment-11336476

**Reply by panla** [reply]
thanks and It has to be said that this setting method is very strange
set it inner fastapi seems better
